src/panels/lovelace/cards/hui-weather-forecast-card.ts

```diff
@@ -0,0 +1,458 @@
+      throw new Error("Invalid Entity");
+    }
+
+    this._config = config;
+  }
+
+  protected updated(changedProps: PropertyValues): void {
+    super.updated(changedProps);
+    if (changedProps.has("hass")) {
+      toggleAttribute(this, "rtl", computeRTL(this.hass!));
+    }
+  }
+
+  protected render(): TemplateResult | void {
+    if (!this._config || !this.hass) {
+      return html``;
+    }
+
+    const stateObj = this.hass.states[this._config.entity];
+
+    if (!stateObj) {
+      return html`
+        <hui-warning
+          >${this.hass.localize(
+            "ui.panel.lovelace.warning.entity_not_found",
+            "entity",
+            this._config.entity
+          )}</hui-warning
+        >
+      `;
+    }
+
+    const forecast = this.computeForecast(stateObj.attributes.forecast);
+
+    return html`
+      <ha-card @click="${this.handleClick}">
+        <div class="header">
+          ${this.computeState(stateObj.state)}
+          <div class="name">${this.computeName(stateObj)}</div>
+        </div>
+        <div class="content">
+          <div class="now">
+            <div class="main">
+              ${this.showWeatherIcon(stateObj.state)
+                ? html`
+                    <ha-icon
+                      icon="${this.getWeatherIcon(stateObj.state)}"
+                    ></ha-icon>
+                  `
+                : ""}
+              <div class="temp">
+                ${stateObj.attributes.temperature}<span
+                  >${this.getUnit("temperature")}</span
+                >
+              </div>
+            </div>
+            <div class="attributes">
+              ${this._showValue(stateObj.attributes.pressure)
+                ? html`
+                    <div>
+                      ${this.hass.localize(
+                        "ui.card.weather.attributes.air_pressure"
+                      )}:
+                      <span class="measurand">
+                        ${stateObj.attributes.pressure}
+                        ${this.getUnit("air_pressure")}
+                      </span>
+                    </div>
+                  `
+                : ""}
+              ${this._showValue(stateObj.attributes.humidity)
+                ? html`
+                    <div>
+                      ${this.hass.localize(
+                        "ui.card.weather.attributes.humidity"
+                      )}:
+                      <span class="measurand"
+                        >${stateObj.attributes.humidity} %</span
+                      >
+                    </div>
+                  `
+                : ""}
+              ${this._showValue(stateObj.attributes.wind_speed)
+                ? html`
+                    <div>
+                      ${this.hass.localize(
+                        "ui.card.weather.attributes.wind_speed"
+                      )}:
+                      <span class="measurand">
+                        ${this.getWindSpeed(stateObj.attributes.wind_speed)}
```

Comment:
These used to be functions because that's how Polymer worked. For readability I would recommend you just inline these functions.

Reply:
Yeah, good call :+1: